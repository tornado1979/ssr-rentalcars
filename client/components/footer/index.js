import React from 'react'

export const Footer = () => {
  return (
    <footer>
      <div className="createdBy">
        <p>Â© 2018 created by theodoreme@gmail.com</p>
      </div>
    </footer>
  )
}
