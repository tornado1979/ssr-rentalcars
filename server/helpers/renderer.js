import React from 'react'
import { renderToString } from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { renderRoutes } from 'react-router-config'
import { Helmet } from 'react-helmet'
import serialize from 'serialize-javascript'

import Routes from '../../client/components/Router/routes'

export default (req, store, context = {}) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter context={context} location={req.path}>
        <div>{renderRoutes(Routes)}</div>
      </StaticRouter>
    </Provider>,
  )
  const helmet = Helmet.renderStatic()

  return (
    `<!DOCTYPE html>
    <html ${helmet.htmlAttributes.toString()}>
      <head>
        <meta charset="UTF-8">
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
      </head>
     <body ${helmet.bodyAttributes.toString()}>
       <div id="root">${content}</div>
       <script>
         window.INITIAL_STATE = ${serialize(store.getState())}
       </script>
       <script src="bundle.js"></script>
     </body>
    </html>`
  )
}
