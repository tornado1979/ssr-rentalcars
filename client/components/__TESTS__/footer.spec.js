import React from 'react'
import renderer from 'react-test-renderer'

import { Footer } from '../footer'

describe('Component footer', () => {
  it('should render correctly', () => {
    expect.assertions(1)
    const component = renderer
      .create(<Footer />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
