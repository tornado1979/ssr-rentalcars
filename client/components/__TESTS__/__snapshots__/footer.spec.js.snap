// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component footer should render correctly 1`] = `
<footer>
  <div
    className="createdBy"
  >
    <p>
      Â© 2018 created by theodoreme@gmail.com
    </p>
  </div>
</footer>
`;
